diff -uNr gnutls-3.6.13/lib/includes/gnutls/gnutls.h.in gnutls-3.6.13-patched/lib/includes/gnutls/gnutls.h.in
--- gnutls-3.6.13/lib/includes/gnutls/gnutls.h.in	2020-03-29 19:24:03.000000000 +0200
+++ gnutls-3.6.13-patched/lib/includes/gnutls/gnutls.h.in	2020-04-29 21:27:20.563461185 +0200
@@ -65,11 +65,7 @@
 #define GNUTLS_CIPHER_RIJNDAEL_CBC GNUTLS_CIPHER_AES_128_CBC
 #define GNUTLS_CIPHER_ARCFOUR GNUTLS_CIPHER_ARCFOUR_128
 
-#if !defined(GNUTLS_INTERNAL_BUILD) && defined(_WIN32)
-# define _SYM_EXPORT __declspec(dllimport)
-#else
-# define _SYM_EXPORT
-#endif
+#define _SYM_EXPORT
 
 #ifdef __GNUC__
 # define __GNUTLS_CONST__  __attribute__((const))
