--- mysql-5.7.15_org/include/my_global.h	2016-08-25 13:52:06.000000000 +0200
+++ mysql-5.7.15/include/my_global.h	2016-09-08 12:18:10.159932092 +0200
@@ -214,7 +214,7 @@
 typedef int	(*qsort_cmp)(const void *,const void *);
 typedef int	(*qsort_cmp2)(const void*, const void *,const void *);
 C_MODE_END
-#ifdef _WIN32
+#if defined(_WIN32) && !defined(__MINGW32__) && !defined(__MINGW64__)
 typedef int       socket_len_t;
 typedef int       sigset_t;
 typedef int       mode_t;
