--- mysql-5.7.15_org/extra/CMakeLists.txt	2016-08-25 13:52:06.000000000 +0200
+++ mysql-5.7.15/extra/CMakeLists.txt	2016-09-12 10:57:36.642545619 +0200
@@ -43,9 +43,10 @@
   )
 ENDIF()
 
+OPTION(COMP_ERR_PATH "Use a custom comp_err." comp_err)
 ADD_CUSTOM_COMMAND(OUTPUT ${PROJECT_BINARY_DIR}/include/mysqld_error.h 
                    ${PROJECT_BINARY_DIR}/sql/share/english/errmsg.sys
-                   COMMAND comp_err
+                   COMMAND ${COMP_ERR_PATH}
                     --charset=${PROJECT_SOURCE_DIR}/sql/share/charsets 
                     --out-dir=${PROJECT_BINARY_DIR}/sql/share/
                     --header_file=${PROJECT_BINARY_DIR}/include/mysqld_error.h 
