diff -uNr TARGET_mysql/mysys/my_error.c TARGET_mysql_patched/mysys/my_error.c
--- TARGET_mysql/mysys/my_error.c	2017-03-18 08:45:14.000000000 +0100
+++ TARGET_mysql_patched/mysys/my_error.c	2017-10-23 16:49:00.388112870 +0200
@@ -114,7 +114,7 @@
         my_snprintf(tmp_buff, sizeof(tmp_buff), 
                     " [OS Error Code : 0x%x]", thr_winerr());
 
-        strcat_s(buf, len, tmp_buff);
+        strcat(buf, tmp_buff);
       }
 
       set_thr_winerr(0);
diff -uNr TARGET_mysql/mysys/stacktrace.c TARGET_mysql_patched/mysys/stacktrace.c
--- TARGET_mysql/mysys/stacktrace.c	2017-10-23 16:01:07.420418709 +0200
+++ TARGET_mysql_patched/mysys/stacktrace.c	2017-10-23 16:48:45.135860355 +0200
@@ -282,10 +282,10 @@
 static void add_to_symbol_path(char *path, size_t path_buffer_size, 
   char *dir, size_t dir_buffer_size)
 {
-  strcat_s(dir, dir_buffer_size, ";");
+  strcat(dir, ";");
   if (!strstr(path, dir))
   {
-    strcat_s(path, path_buffer_size, dir);
+    strcat(path, dir);
   }
 }
 
@@ -316,7 +316,7 @@
   if(p)
   { 
     *p= 0;
-    strcat_s(pdb_debug_dir, sizeof(pdb_debug_dir), "\\debug;");
+    strcat(pdb_debug_dir, "\\debug;");
     add_to_symbol_path(path, size, pdb_debug_dir, sizeof(pdb_debug_dir));
   }
 #endif
@@ -358,7 +358,7 @@
   envvar= getenv("_NT_SYMBOL_PATH");
   if(envvar)
   {
-    strcat_s(path, size, envvar);
+    strcat(path, envvar);
   }
 }
 
