--- mysql-5.7.15_org/mysys/my_thr_init.c	2016-08-25 13:52:06.000000000 +0200
+++ mysql-5.7.15/mysys/my_thr_init.c	2016-09-08 13:44:14.170165725 +0200
@@ -298,7 +298,7 @@
   if (!my_thread_global_init_done)
     return TRUE; /* cannot proceed with unintialized library */
 
-#ifdef _WIN32
+#if defined(_WIN32) && !defined(__MINGW32__) && !defined(__MINGW64__)
   install_sigabrt_handler();
 #endif
 
@@ -435,7 +435,7 @@
 #endif /* DBUG_OFF */
 
 
-#ifdef _WIN32
+#if defined(_WIN32) && !defined(__MINGW32__) && !defined(__MINGW64__)
 /*
   In Visual Studio 2005 and later, default SIGABRT handler will overwrite
   any unhandled exception filter set by the application  and will try to
