--- mysql-5.7.15_org/mysys/my_winfile.c	2016-08-25 13:52:06.000000000 +0200
+++ mysql-5.7.15/mysys/my_winfile.c	2016-09-08 15:12:31.683595002 +0200
@@ -44,9 +44,10 @@
   here are useful only in scernarios that use low-level IO with my_win_fileno()
 */
 
-#ifdef _WIN32
+#if defined(_WIN32)
 
 #include "mysys_priv.h"
+#include "my_sys.h"
 #include "my_thread_local.h"
 #include <share.h>
 #include <sys/stat.h>
