diff -uNr mysql-5.7.15/cmake/build_configurations/compiler_options.cmake mysql-5.7.15_patched/cmake/build_configurations/compiler_options.cmake
--- mysql-5.7.15/cmake/build_configurations/compiler_options.cmake	2016-08-25 13:52:06.000000000 +0200
+++ mysql-5.7.15_patched/cmake/build_configurations/compiler_options.cmake	2016-09-07 21:53:51.000000000 +0200
@@ -15,7 +15,7 @@
 
 INCLUDE(CheckCCompilerFlag)
 INCLUDE(CheckCXXCompilerFlag)
-INCLUDE(cmake/floating_point.cmake)
+#INCLUDE(cmake/floating_point.cmake)
 
 IF(SIZEOF_VOIDP EQUAL 4)
   SET(32BIT 1)

