--- mysql-5.7.15_org/include/m_string.h	2016-08-25 13:52:06.000000000 +0200
+++ mysql-5.7.15/include/m_string.h	2016-09-08 12:33:20.831014353 +0200
@@ -129,7 +129,7 @@
 
 static inline char *my_strtok_r(char *str, const char *delim, char **saveptr)
 {
-#if defined _WIN32
+#if defined(_WIN32) && !defined(__MINGW32__) && !defined(__MINGW64__)
   return strtok_s(str, delim, saveptr);
 #else
   return strtok_r(str, delim, saveptr);
