diff -uNr mysql-5.7.15/extra/yassl/src/yassl_int.cpp mysql-5.7.15_patched/extra/yassl/src/yassl_int.cpp
--- mysql-5.7.15/extra/yassl/src/yassl_int.cpp	2016-08-25 13:52:06.000000000 +0200
+++ mysql-5.7.15_patched/extra/yassl/src/yassl_int.cpp	2016-09-07 21:57:25.000000000 +0200
@@ -20,7 +20,7 @@
 // First include (the generated) my_config.h, to get correct platform defines.
 #include "my_config.h"
 #ifdef _WIN32
-#include<Windows.h>
+#include <windows.h>
 #else
 #include <pthread.h>
 #endif
diff -uNr mysql-5.7.15/include/mysql/psi/mysql_socket.h mysql-5.7.15_patched/include/mysql/psi/mysql_socket.h
--- mysql-5.7.15/include/mysql/psi/mysql_socket.h	2016-08-25 13:52:06.000000000 +0200
+++ mysql-5.7.15_patched/include/mysql/psi/mysql_socket.h	2016-09-07 22:34:14.000000000 +0200
@@ -29,7 +29,7 @@
 #ifdef _WIN32
   #include <ws2def.h>
   #include <winsock2.h>
-  #include <MSWSock.h>
+  #include <mswsock.h>
   #define SOCKBUF_T char
 #else
   #include <netinet/in.h>
diff -uNr mysql-5.7.15/libbinlogevents/include/binlog_event.h mysql-5.7.15_patched/libbinlogevents/include/binlog_event.h
--- mysql-5.7.15/libbinlogevents/include/binlog_event.h	2016-08-25 13:52:06.000000000 +0200
+++ mysql-5.7.15_patched/libbinlogevents/include/binlog_event.h	2016-09-08 09:52:29.721204500 +0200
@@ -41,7 +41,7 @@
 #include <climits>
 
 #if defined(_WIN32)
-#include <Winsock2.h>
+#include <winsock2.h>
 #else
 #include <sys/times.h>
 #endif
diff -uNr mysql-5.7.15/mysys/my_gethwaddr.c mysql-5.7.15_patched/mysys/my_gethwaddr.c
--- mysql-5.7.15/mysys/my_gethwaddr.c	2016-08-25 13:52:06.000000000 +0200
+++ mysql-5.7.15_patched/mysys/my_gethwaddr.c	2016-09-07 22:02:14.000000000 +0200
@@ -138,6 +138,7 @@
 #define VOID void
 #endif
 
+#include <wincrypt.h>
 #include <iphlpapi.h>
 
 /*

