diff -uNr mysql-5.7.15/CMakeLists.txt mysql-5.7.15_patched/CMakeLists.txt
--- mysql-5.7.15/CMakeLists.txt	2016-08-25 13:52:06.000000000 +0200
+++ mysql-5.7.15_patched/CMakeLists.txt	2016-09-08 10:13:11.840222483 +0200
@@ -585,8 +585,8 @@
 ADD_SUBDIRECTORY(mysys)
 ADD_SUBDIRECTORY(mysys_ssl)
 ADD_SUBDIRECTORY(libmysql)
-ADD_SUBDIRECTORY(libbinlogevents)
-ADD_SUBDIRECTORY(libbinlogstandalone)
+#ADD_SUBDIRECTORY(libbinlogevents)
+#ADD_SUBDIRECTORY(libbinlogstandalone)
 
 IF(WITH_UNIT_TESTS)
   # Visual Studio 11 needs this extra flag in order to compile gmock.
@@ -608,9 +608,9 @@
 ENDIF()
 
 ADD_SUBDIRECTORY(extra)
-ADD_SUBDIRECTORY(client)
-ADD_SUBDIRECTORY(sql/share)
-ADD_SUBDIRECTORY(libservices)
+#ADD_SUBDIRECTORY(client)
+#ADD_SUBDIRECTORY(sql/share)
+#ADD_SUBDIRECTORY(libservices)
 
 IF(UNIX)
   ADD_SUBDIRECTORY(man)
@@ -628,7 +628,7 @@
 
 # scripts/mysql_config depends on client and server targets loaded above.
 # It is referenced by some of the directories below, so we insert it here.
-ADD_SUBDIRECTORY(scripts)
+#ADD_SUBDIRECTORY(scripts)
 
 IF(NOT WITHOUT_SERVER)
   ADD_SUBDIRECTORY(mysql-test)
@@ -679,8 +679,8 @@
 ELSE()
   SET(CPACK_GENERATOR "TGZ")
 ENDIF() 
-ADD_SUBDIRECTORY(packaging/WiX)
-ADD_SUBDIRECTORY(packaging/solaris)
+#ADD_SUBDIRECTORY(packaging/WiX)
+#ADD_SUBDIRECTORY(packaging/solaris)
 
 IF(UNIX)
   INSTALL(FILES Docs/mysql.info DESTINATION ${INSTALL_INFODIR} OPTIONAL COMPONENT Info)

