--- mysql-5.7.18/cmake/build_configurations/compiler_options.cmake
+++ mysql-5.7.18_patched/cmake/build_configurations/compiler_options.cmake
@@ -15,8 +15,8 @@
 
 INCLUDE(CheckCCompilerFlag)
 INCLUDE(CheckCXXCompilerFlag)
-INCLUDE(cmake/compiler_bugs.cmake)
-INCLUDE(cmake/floating_point.cmake)
+#INCLUDE(cmake/floating_point.cmake)
+#INCLUDE(cmake/compiler_bugs.cmake)
 
 IF(SIZEOF_VOIDP EQUAL 4)
   SET(32BIT 1)

