--- mysql-5.7.15_org/CMakeLists.txt	2016-08-25 13:52:06.000000000 +0200
+++ mysql-5.7.15/CMakeLists.txt	2016-09-08 12:25:09.410925883 +0200
@@ -147,6 +147,10 @@
 SET(MYSQL_MACHINE_TYPE ${CMAKE_SYSTEM_PROCESSOR})
 
 
+IF((${CMAKE_SYSTEM_NAME} STREQUAL "Windows") AND (${CMAKE_COMPILER_IS_GNUCC}))
+  ADD_DEFINITIONS(-D_POSIX)
+ENDIF((${CMAKE_SYSTEM_NAME} STREQUAL "Windows") AND (${CMAKE_COMPILER_IS_GNUCC}))
+
 # Include the platform-specific file. To allow exceptions, this code
 # looks for files in order of how specific they are. If there is, for
 # example, a generic Linux.cmake and a version-specific
