diff -uNr mariadb-connector-c-3.1.13-src/CMakeLists.txt mariadb-connector-c-3.1.13-src_patched/CMakeLists.txt
--- mariadb-connector-c-3.1.13-src/CMakeLists.txt	2022-04-20 12:05:11.074851933 +0200
+++ mariadb-connector-c-3.1.13-src_patched/CMakeLists.txt	2022-04-20 12:06:20.266878517 +0200
@@ -283,6 +283,7 @@
       SET(SSL_LIBRARIES ${OPENSSL_SSL_LIBRARY} ${OPENSSL_CRYPTO_LIBRARY})
       IF(WIN32)
         CHECK_INCLUDE_FILES (${OPENSSL_INCLUDE_DIR}/openssl/applink.c HAVE_OPENSSL_APPLINK_C)
+        SET(SSL_LIBRARIES ${SSL_LIBRARIES} crypt32 ws2_32)
       ENDIF()
       INCLUDE_DIRECTORIES(BEFORE ${OPENSSL_INCLUDE_DIR})
 
