diff -uNr mariadb-connector-c-3.1.7-src/plugins/auth/CMakeLists.txt mariadb-connector-c-3.1.7-src-patched/plugins/auth/CMakeLists.txt
--- mariadb-connector-c-3.1.7-src/plugins/auth/CMakeLists.txt	2020-04-16 08:47:51.287741105 +0200
+++ mariadb-connector-c-3.1.7-src-patched/plugins/auth/CMakeLists.txt	2020-04-16 08:51:31.127652836 +0200
@@ -30,7 +30,7 @@
 REGISTER_PLUGIN(TARGET dialog
                 TYPE MARIADB_CLIENT_PLUGIN_AUTH
                 CONFIGURATIONS DYNAMIC STATIC OFF
-                DEFAULT DYNAMIC
+                DEFAULT STATIC
                 SOURCES ${CC_SOURCE_DIR}/plugins/auth/dialog.c
                         ${CC_SOURCE_DIR}/libmariadb/get_password.c)
 
@@ -54,7 +54,7 @@
   REGISTER_PLUGIN(TARGET client_ed25519
                 TYPE MARIADB_CLIENT_PLUGIN_AUTH
                 CONFIGURATIONS DYNAMIC STATIC OFF
-                DEFAULT DYNAMIC
+                DEFAULT OFF
                 SOURCES ${CC_SOURCE_DIR}/plugins/auth/ed25519.c 
                         ${REF10_SOURCES}
                         ${CRYPT_SOURCE}
@@ -70,7 +70,7 @@
   REGISTER_PLUGIN(TARGET caching_sha2_password
                 TYPE MARIADB_CLIENT_PLUGIN_AUTH
                 CONFIGURATIONS DYNAMIC STATIC OFF
-                DEFAULT DYNAMIC
+                DEFAULT STATIC
                 SOURCES ${CC_SOURCE_DIR}/plugins/auth/caching_sha2_pw.c 
                         ${CRYPT_SOURCE}
                 LIBRARIES ${CRYPT_LIBS})
@@ -81,7 +81,7 @@
     REGISTER_PLUGIN(TARGET sha256_password
                     TYPE MARIADB_CLIENT_PLUGIN_AUTH
                     CONFIGURATIONS DYNAMIC STATIC OFF
-                    DEFAULT DYNAMIC
+                    DEFAULT STATIC
                     SOURCES ${AUTH_DIR}/sha256_pw.c
                     LIBRARIES ${CRYPT_LIBS})
   ENDIF()
@@ -106,7 +106,7 @@
   REGISTER_PLUGIN(TARGET auth_gssapi_client
                   TYPE MARIADB_CLIENT_PLUGIN_AUTH
                   CONFIGURATIONS DYNAMIC STATIC OFF
-                  DEFAULT DYNAMIC
+                  DEFAULT OFF
                   SOURCES ${GSSAPI_SOURCES}
                   INCLUDES ${CC_SOURCE_DIR}/plugins/auth ${GSSAPI_INCS}
                   LIBRARIES ${GSSAPI_LIBS})
@@ -123,6 +123,6 @@
 REGISTER_PLUGIN(TARGET mysql_clear_password
                 TYPE MARIADB_CLIENT_PLUGIN_AUTH
                 CONFIGURATIONS DYNAMIC STATIC OFF
-                DEFAULT DYNAMIC
+                DEFAULT STATIC
                 SOURCES ${AUTH_DIR}/mariadb_cleartext.c)
 
diff -uNr mariadb-connector-c-3.1.7-src/plugins/io/CMakeLists.txt mariadb-connector-c-3.1.7-src-patched/plugins/io/CMakeLists.txt
--- mariadb-connector-c-3.1.7-src/plugins/io/CMakeLists.txt	2020-04-16 08:47:51.291741102 +0200
+++ mariadb-connector-c-3.1.7-src-patched/plugins/io/CMakeLists.txt	2020-04-16 08:52:07.307639273 +0200
@@ -7,7 +7,7 @@
     REGISTER_PLUGIN(TARGET remote_io
                   TYPE MARIADB_CLIENT_PLUGIN_IO
                   CONFIGURATIONS DYNAMIC STATIC OFF
-                  DEFAULT DYNAMIC
+                  DEFAULT OFF
                   SOURCES ${CC_SOURCE_DIR}/plugins/io/remote_io.c
                   INCLUDES ${CURL_INCLUDE_DIR}
                   LIBRARIES ${CURL_LIBRARIES})
diff -uNr mariadb-connector-c-3.1.7-src/plugins/pvio/CMakeLists.txt mariadb-connector-c-3.1.7-src-patched/plugins/pvio/CMakeLists.txt
--- mariadb-connector-c-3.1.7-src/plugins/pvio/CMakeLists.txt	2020-04-16 08:47:51.291741102 +0200
+++ mariadb-connector-c-3.1.7-src-patched/plugins/pvio/CMakeLists.txt	2020-04-16 08:52:25.799632426 +0200
@@ -15,13 +15,13 @@
   REGISTER_PLUGIN(TARGET pvio_npipe
                 TYPE MARIADB_CLIENT_PLUGIN_PVIO
                 CONFIGURATIONS STATIC DYNAMIC DEFAULT
-                DEFAULT DYNAMIC
+                DEFAULT STATIC
                 SOURCES ${CC_SOURCE_DIR}/plugins/pvio/pvio_npipe.c)
 
   # shared memory
   REGISTER_PLUGIN(TARGET pvio_shmem
                 TYPE MARIADB_CLIENT_PLUGIN_PVIO
                 CONFIGURATIONS STATIC DYNAMIC DEFAULT
-                DEFAULT DYNAMIC
+                DEFAULT STATIC
                 SOURCES ${CC_SOURCE_DIR}/plugins/pvio/pvio_shmem.c)
 ENDIF()
