diff -uNr mariadb-connector-c-3.1.13-src/CMakeLists.txt mariadb-connector-c-3.1.13-src_patched/CMakeLists.txt
--- mariadb-connector-c-3.1.13-src/CMakeLists.txt	2022-04-20 12:08:37.270925957 +0200
+++ mariadb-connector-c-3.1.13-src_patched/CMakeLists.txt	2022-04-20 12:08:53.590931212 +0200
@@ -365,7 +365,7 @@
   ENDIF()
 ENDIF()
 IF(WITH_SSL)
-  SET(SYSTEM_LIBS ${SYSTEM_LIBS} ${SSL_LIBRARIES})
+  SET(SYSTEM_LIBS ${SYSTEM_LIBS} ${SSL_LIBRARIES} ${LIBPTHREAD})
 ENDIF()
 MESSAGE1(SYSTEM_LIBS "SYSTEM_LIBS ${SYSTEM_LIBS}")
 MARK_AS_ADVANCED(SYSTEM_LIBS)
