diff -uNr mariadb-connector-c-3.1.13-src/CMakeLists.txt mariadb-connector-c-3.1.13-src_patched/CMakeLists.txt
--- mariadb-connector-c-3.1.13-src/CMakeLists.txt	2022-04-20 12:09:26.998941732 +0200
+++ mariadb-connector-c-3.1.13-src_patched/CMakeLists.txt	2022-04-20 12:09:42.018946360 +0200
@@ -308,7 +308,10 @@
     IF(GNUTLS_FOUND)
       ADD_DEFINITIONS(-DHAVE_GNUTLS -DHAVE_TLS)
       SET(SSL_SOURCES "${CC_SOURCE_DIR}/libmariadb/secure/gnutls.c")
-      SET(SSL_LIBRARIES ${GNUTLS_LIBRARY})
+      SET(SSL_LIBRARIES ${GNUTLS_LIBRARY} ${HOGWEED_LIBRARY} ${NETTLE_LIBRARY} ${GMP_LIBRARY})
+      IF(WIN32)
+        SET(SSL_LIBRARIES ${SSL_LIBRARIES} crypt32)
+      ENDIF(WIN32)
       SET(TLS_LIBRARY_VERSION "GnuTLS ${GNUTLS_VERSION_STRING}")
       INCLUDE_DIRECTORIES(${GNUTLS_INCLUDE_DIR})
     ELSE()
