diff -uNr mariadb-connector-c-3.1.7-src/CMakeLists.txt mariadb-connector-c-3.1.7-src-patched/CMakeLists.txt
--- mariadb-connector-c-3.1.7-src/CMakeLists.txt	2020-04-16 08:54:37.671519639 +0200
+++ mariadb-connector-c-3.1.7-src-patched/CMakeLists.txt	2020-04-16 08:54:47.815486432 +0200
@@ -321,7 +321,10 @@
     IF(GNUTLS_FOUND)
       ADD_DEFINITIONS(-DHAVE_GNUTLS -DHAVE_TLS)
       SET(SSL_SOURCES "${CC_SOURCE_DIR}/libmariadb/secure/gnutls.c")
-      SET(SSL_LIBRARIES ${GNUTLS_LIBRARY})
+      SET(SSL_LIBRARIES ${GNUTLS_LIBRARY} ${HOGWEED_LIBRARY} ${NETTLE_LIBRARY} ${GMP_LIBRARY})
+      IF(WIN32)
+        SET(SSL_LIBRARIES ${SSL_LIBRARIES} crypt32)
+      ENDIF(WIN32)
       SET(TLS_LIBRARY_VERSION "GnuTLS ${GNUTLS_VERSION_STRING}")
       INCLUDE_DIRECTORIES(${GNUTLS_INCLUDE_DIR})
     ELSE()
